### ISSUE TITLE
DeleteFromQuery raises System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

### ISSUE THREAD

**Issue by HaraldDobner** [zzzprojects/EntityFramework-Extensions#163]
Hello ZZZ team,
I have a problem integrating Entity Framework Extentions into a Repository Pattern. Here at I get the error message "The given key was not present in the dictionary." It comes from inside the EF Extentions, so I can not get there.

But first things first.

We want to build a new project. The basis for this project is aspnetboilerplate.com and the paid extension aspnetzero.com.
Now EntityFramework Integration is abstracted with the RepositoryPattern. The Repository Pattern is included into boilerplate and can not be easily removed.

Now we have set up our own application repository, which derives from the boilerplate repository implementation. There I have direct access to the DbContext. The bulk functions work perfectly, only with the methods DeleteFromQuery () and UpdateFromQuery (...) do I get an exception of the type System.Collections.Generic.KeyNotFoundException

The associated StackTrace is:

```
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.ThrowHelper.ThrowKeyNotFoundException ()
   at System.Collections.Generic.Dictionary`2.get_Item (TKey key)
   at Z.EntityFramework.Extensions.BatchDelete.CreateCommand (IQueryable query, IEntityType entity)
   at Z.EntityFramework.Extensions.BatchDelete.Execute [T] (IQueryable`1 query)
   at B4P.Sexpress.EntityFrameworkCore.Repositories.B4SepaClearerRepository.DeleteFromQuery (Expression`1 deleteConditionFunc)
   at Castle.DynamicProxy.AbstractInvocation.Proceed ()
   at Abp.Domain.Uow.UnitOfWorkInterceptor.PerformSyncUow (IInvocation invocation, UnitOfWorkOptions options)
   at Castle.DynamicProxy.AbstractInvocation.Proceed ()
   at B4P.Sexpress.Jobs.TestJob.Execute ()
```

I'm stuck here because I can not debug in the implementation. Can you help me?

The system is as follows:
Boilerplate 3.9.0
Entity Framework Core 2.1.3
Z.EntityFramework.Extentions.EFCore v2.1.35
Ms SqlServer 14.0.2002.14 on a develpoer machine.

Below are the calls
`this.Context.SepaClearers.Where (x => x.Sdd == true) .DeleteFromQuery ();`

or

`this.Context.SepaClearers.Where (x => x.Sdd == true) .UpdateFromQuery (x => new SepaClearer {BankName = "SepaBank"});`

So basically it's the same thing in the documentary.
If I try it out in a console application that contains nothing but the EntityFramework and the extentions, it works just fine.

Can you help me?

**Comment by HaraldDobner** [comment]
Hello Jonathan,

yes, that was the problem. Now it works perfect.

Boilerplate really has a system for automatic multi-tenancy support, as described in the microsoft link.

Thank you for your superfast help. If you could add support for Global Query Filters, it would be super cool.

Thank you again.

Best Regands,

Harald

**Comment by JonathanMagnan** [comment]
Yup,

We are already on it. We made some progress yesterday by finding how `EF Core` support them. Only a small puzzle piece is missing now before we start supporting it ;)

**Comment by JonathanMagnan** [comment]
Hello @HaraldDobner ,

It looks that we never successfully make it fully works.

We will close the issue here but we will keep it open in our internal issue tracker.

We will for sure try it from time to time when the `Entity Framework` team release new version.

At this moment, the only way to make it works is by ignoring with `IgnoreQueryFilters()`.

Best Regards,

Jonathan

**Comment by ThomasAy** [comment]
Hello,

Any update on this? I'm using `UpdateFromQuery` on Entity Framework and got the same `KeyNotFoundException` exception.

**Comment by JonathanMagnan** [comment]
Hello @ThomasAy ,

The solution is currently the same as my last post: https://github.com/zzzprojects/EntityFramework-Extensions/issues/163#issuecomment-458563160

Best Regards,

Jon

**Comment by ThomasAy** [comment]
@JonathanMagnan the given solution is for EF Core while I'm using EF6. I don't see how it applies to me then. Could you elaborate?

**Comment by JonathanMagnan** [comment]
Hello @ThomasAy ,

In this case, I recommend you create another issue and provide a runnable project with the problem. It doesn’t need to be your project, just a new solution with the minimum code to reproduce the issue.

Since your problem is different (under EF6), I believe it will be better for people searching for it to have a problem/solution inside a different issue as well.

Best Regards,

Jon
